### ISSUE TITLE
Use 'docker compose' instead of 'docker-compose'

### ISSUE THREAD

**Issue by stewartbryson** [avast/gradle-docker-compose-plugin#322]
Have you considered porting this to use the new compose functionality in the Docker daemon?

Thank you.

**Comment by henrik242** [comment]
Can this be used as a workaround? ```
$ echo '#!/usr/bin/env bash

docker compose $@' > ~/bin/docker-compose
$ chmod 755 ~/bin/docker-compose
$ which docker-compose
/Users/me/bin/docker-compose
$ docker-compose version
Docker Compose version v2.3.3
```

**Comment by henrik242** [comment]
I've tried to add support in the PR at #348. Might be too simplistic, but at least shouldn't break anything with the default config.

**Comment by augi** [comment]
@henrik242 Actually, there is `compose-switch` utility that works like a _bridge_ between `docker-compose` and `docker compose`: https://github.com/docker/compose-switch/

**Comment by henrik242** [comment]
`compose-switch` requires a separate install, which we'd rather not have to push on all users of our project.

The default bundled `docker-compose` in Docker Desktop is v1.29.2 from may 2021, and doesn't support the latest [compose-spec](https://github.com/compose-spec/compose-spec/blob/master/spec.md)

```
$ ls -l /usr/local/bin/docker-compose
lrwxr-xr-x  1 root  admin  62 25 aug  2021 /usr/local/bin/docker-compose -> /Applications/Docker.app/Contents/Resources/bin/docker-compose
$  ls -l /usr/local/bin/docker
lrwxr-xr-x  1 root  admin  54 17 nov  2020 /usr/local/bin/docker -> /Applications/Docker.app/Contents/Resources/bin/docker
$ /usr/local/bin/docker-compose version
docker-compose version 1.29.2, build 5becea4c
docker-py version: 5.0.0
CPython version: 3.9.0
OpenSSL version: OpenSSL 1.1.1h  22 Sep 2020
$ /usr/local/bin/docker compose version
Docker Compose version v2.3.3
```

**Comment by augi** [comment]
I didn't mean `compose-switch` as replacement for a feature of this plugin, I just wanted to mention it as an alternative for the solution proposed by you 👌

**Comment by augi** [comment]
Imagine we introduced the proposed `ComposeV2` switch - then if enabled, you would have to force all the users of your project to switch to a latest version of Docker Desktop that supports `docker compose`.

What would be the perfect solution would be not to introduce the `ComposeV2` switch at all, but make the decision internally, if use `docker-compose` or `docker compose`. But I'm not sure how to detect that Compose V2 should be used 🤔

**Comment by henrik242** [comment]
You could of course check versions of both `docker-compose` and `docker compose` and pick the latest one, but debugging errors could be tricky if the user is not aware of what is happening.

I'd rather have a manual switch for now.

**Comment by biancarosa** [comment]
I just had to spend some extra time making this work as "docker-compose" is not an official alias for "docker compose" in the main installation. 

Could there be a config variable saying "useComposeV2" that can, eventually, have its default set to true? That'd require maintaning both codes for a little while, though.

**Comment by bassco** [comment]
Our docker-compose version switched from 1.29.2 to 2.10.2 on our build images and we are getting invalid tag reference names due to the service name prefixes changing to `-` to `_` in our CI environment.

```console
> Task :composeBuild FAILED
invalid tag "6a186a353405cef76803929dcf99538d_ads-decision-engine_-localstack": invalid reference format

FAILURE: Build failed with an exception.
```

Anyone have a workaround that works with the new docker-compose version 2 or do I need to get the platform team to revert the docker-compose version?

**Comment by bassco** [comment]
If anyone comes across the above issue, the workaround is to set the following env var `COMPOSE_COMPATIBILITY=1` which restores the v1 service name prefix.

**Comment by robertfmurdock** [comment]
Looks like the time is coming soon: warnings in the latest update of docker desktop.

https://docs.docker.com/compose/migrate/

**Comment by augi** [comment]
Thanks. Feel free to prepare a new PR for this feature 🙏

**Comment by rt-works** [comment]
we also can take a look at the continiously supported solution from the latest spring-boot 3.1 org.springframework.boot:spring-boot-docker-compose 
https://tpbabparn.medium.com/spring-boot-3-1-integrated-docker-compose-on-development-environment-spring-webflux-example-1ddcbfe052f

**Comment by jvwilge** [comment]
Won't this tie you to the Spring eco system? The nice thing of the avast gradle-docker-compose-plugin for me is that it is quite standalone and has a clear boundary between code and Docker, with Spring it usually gets blurry and/or you need many dependencies.

**Comment by henrik242** [comment]
@rt-works You shouldn't quote `$@`

**Comment by rt-works** [comment]
worked for me with quoting

**Comment by henrik242** [comment]
@rt-works For simple commands, yeah. With more options it might not.

**Comment by rt-works** [comment]
Good to know :) For the "up -f docker-compose.yaml" which is passed there by gradle plugin that was sufficient.

**Comment by joecotton-wk** [comment]
For those interested, I've opened a [PR](https://github.com/avast/gradle-docker-compose-plugin/pull/410) to add this feature to the plugin. If anyone has time to do additional testing or to review, it would be much appreciated.

**Comment by augi** [comment]
Fixed by #410
